### DISCUSSION TITLE
display-line-numbers-mode reactivated in pdf-view-mode when previewing (consult-buffer)

### DISCUSSION THREAD

**Discussion by maxecharel** [minad/consult#853] [Q&A] [answered]
Hi,
In my init file I have `global-display-line-numbers-mode` set to `t`. I nonetheless deactivate `display-line-numbers-mode` in `pdf-view-mode` via hooks: `(add-hook 'pdf-view-mode-hook (lambda () (display-line-numbers-mode -1)))`

When previewing files with `consult-buffer` (great functionality btw), it appears that `pdf-view-mode-hook` is not considered and I get the warning message:
```
Warning (pdf-view): These modes are incompatible with `pdf-view-mode',
	please deactivate them (or customize pdf-view-incompatible-modes): display-line-numbers-mode
```
After having checked the documentation of `consult`, I thought a priori that `consult-preview-allowed-hooks` could constitute a workaround. Unfortunately (after having thought twice and even tried to implement it), such a workaround requires to add a hook to `find-file-hook` which in turn keeps `display-line-numbers-mode`  disabled when files (PDF or not, the latter being the problem) are actually opened.

Have I missed something? Is there another way to tell `consult-buffer` to deactivate `display-line-numbers-mode` when previewing PDF files?

Thank you for the amazing work!

Emacs 29.1
Consult 20230919.1954

**Comment by minad** [comment]
As an alternative you could enable display-line-numbers mode per mode (text-mode/prog-mode/conf-mode), then the issue should not occur.

**Reply by maxecharel** [reply]
Thanks for the suggestion (and sorry for the mischaracterization of the post), I thought about that but given that I only disable this mode for `pdf-view-` and `term-mode` I wondered if there was a way to just tell `consult-buffer` to comply to `pdf-view-mode-hook`. But if there is not I am just going to specify for which modes `display-line-numbers` should be enabled. Thanks.

**Comment by maxecharel** [comment]
For users facing the same "issue", as previously suggested something like
``` 
(dolist (hook '(prog-mode-hook text-mode-hook conf-mode-hook))
    (add-hook hook 'display-line-numbers-mode))
``` 
solves the problem. However, such workaround leads `consult-buffer` preview not to display line numbers for non-PDF files anymore (obviously for PDF files it is the desired behavior).  While it is highly likely that it will not constitute a disturbance for a large majority of users, I put that here for 'exhaustiveness'. Thanks again for the great work @minad

**Reply by minad** [reply]
If you still want to use the global mode such that line numbers are also displayed during preview you should configure the global mode such that pdf-view is excluded, instead of enabling and then disabling it locally again. I haven't looked at the global-display-line-number-mode configuration but this should be possible.

**Comment by maxecharel** [comment, marked as answer]
@minad good idea, I've looked further into it. At the minor price of redefining a function of `display-line-numbers.el` used in the definition of the globalized minor mode , the following completely solves the 'issue':
```elisp
  (require 'display-line-numbers)
  (defun display-line-numbers--turn-on ()
    "Turn on `display-line-numbers-mode'."
    (unless (or (minibufferp) (eq major-mode 'pdf-view-mode))
      (display-line-numbers-mode)))
```
Now one can simply set ` (global-display-line-numbers-mode 1)` and the original issue is solved while still having line numbers in the preview of non-PDF files.

**Reply by maxecharel** [reply]
Note that in this quick example I've only focused on `pdf-view-mode` but `term-mode` or other modes can easily be added.

**Reply by maxecharel** [reply]
@minad thanks a lot (again) :)

**Reply by minad** [reply]
I think you should open a feature request upstream. It should be possible to configure global-display-mode via a customization variable `global-display-line-numbers-modes` generated via the `:predicate` argument of `define-globalized-minor-mode`.

**Reply by lemquy** [reply]
Do we need to add the function into a certain hook?
